packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js

```diff
@@ -1779,7 +1779,7 @@ function isNodeLike(val) {
 
 function isSameIdentifier(a, b) {
   return (
-    a.type === 'Identifier' &&
+    (a.type === 'Identifier' || a.type === 'JSXIdentifier') &&
```

Comment:
That is a bit hacky since it would return true when comparing `isSameIdentifier(Component, <Component />)`. Can't imagine a path that would lead to this though.

Reply:
That sounds odd. Can we find a way to tighten it up?

Reply:
Added a check that makes sure `a` and `b` are the same kind of identifier. `isSameIdentifier-ish` is probably more accurate at this point.